apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/graphql/ast/document.kt

```diff
@@ -0,0 +1,128 @@
+package com.apollographql.apollo.compiler.parser.graphql.ast
+
+import com.apollographql.apollo.compiler.parser.antlr.GraphQLLexer
+import com.apollographql.apollo.compiler.parser.antlr.GraphQLParser
+import com.apollographql.apollo.compiler.parser.error.DocumentParseException
+import com.apollographql.apollo.compiler.parser.error.ParseException
+import okio.Buffer
+import okio.BufferedSink
+import okio.buffer
+import okio.sink
+import org.antlr.v4.runtime.BaseErrorListener
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+import org.antlr.v4.runtime.RecognitionException
+import org.antlr.v4.runtime.Recognizer
+import org.antlr.v4.runtime.Token
+import org.antlr.v4.runtime.atn.PredictionMode
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
+
+fun GQLDocument.withBuiltinTypes(): GQLDocument {
+  val buildInsInputStream = javaClass.getResourceAsStream("/builtins.sdl")
+  return copy(
+      definitions = definitions + GQLDocument.parseInternal(buildInsInputStream).definitions
+  )
+}
+
+fun GQLDocument.withoutBuiltinTypes(): GQLDocument {
+  return copy(
+      definitions = definitions.filter {
+        when ((it as? GQLNamed)?.name) {
+          "Int", "Float", "String", "ID", "Boolean", "__Schema",
+          "__Type", "__Field", "__InputValue", "__EnumValue", "__TypeKind",
+          "__Directive", "__DirectiveLocation" -> false
+          else -> true
```

Comment:
nit: can it be shorter `.filterNot { (it as? GQLNamed)?.name in listOf(....) }`

Reply:
Rewrote it in #2756 : https://github.com/apollographql/apollo-android/blob/ec9190dd77d6421cd7878dc3bc86ab83e7aa6f9f/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/gql/gqltypedefinition.kt#L25